cmd/beaker/main.go

```diff
@@ -1,76 +1,151 @@
 package main
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"net/http"
 	"os"
+	"os/signal"
+	"strings"
 
-	"github.com/fatih/color"
-	kingpin "gopkg.in/alecthomas/kingpin.v2"
-
-	"github.com/allenai/beaker/cmd/beaker/cluster"
-	configCmd "github.com/allenai/beaker/cmd/beaker/config"
-	"github.com/allenai/beaker/cmd/beaker/dataset"
-	"github.com/allenai/beaker/cmd/beaker/experiment"
-	"github.com/allenai/beaker/cmd/beaker/group"
-	"github.com/allenai/beaker/cmd/beaker/image"
-	"github.com/allenai/beaker/cmd/beaker/options"
-	"github.com/allenai/beaker/cmd/beaker/secret"
-	"github.com/allenai/beaker/cmd/beaker/task"
-	"github.com/allenai/beaker/cmd/beaker/workspace"
 	"github.com/allenai/beaker/config"
+	"github.com/beaker/client/api"
+	"github.com/beaker/client/client"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+// These variables are set externally by the linker.
+var (
+	version = "dev"
+	commit  = "unknown"
+)
+
+var beaker *client.Client
+var beakerConfig *config.Config
+var ctx context.Context
+var quiet bool
+var format string
+
+const (
+	formatJSON = "json"
 )
 
 func main() {
+	var cancel context.CancelFunc
+	ctx, cancel = withSignal(context.Background())
+	defer cancel()
+
 	errorPrefix := color.RedString("Error:")
 
-	config, err := config.New()
+	var err error
+	beakerConfig, err = config.New()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%s %+v\n", errorPrefix, err)
 		os.Exit(1)
 	}
 
-	if opts, err := newApp(config); err != nil {
-		if opts.Debug {
-			fmt.Fprintf(os.Stderr, "%s %+v\n", errorPrefix, err)
+	beaker, err = client.NewClient(beakerConfig.BeakerAddress, beakerConfig.UserToken)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%s %+v\n", errorPrefix, err)
+		os.Exit(1)
+	}
+
+	root := &cobra.Command{
+		Use:   "beaker",
+		Short: "Beaker is a tool for running machine learning experiments.",
+		// TODO What do these do?
+		// SilenceUsage: true,
+		// SilenceErrors: true,
```

Comment:
SilenceUsage prevents printing usage on every error.  We want this one set to true.
SilenceErrors prevents printing parse errors. Since we print returned errors ourselves, including parse errors, we also want this one set to true.

```suggestion
		SilenceUsage: true,
		SilenceErrors: true,
```

Reply:
Turned these on and added error printing on `root.Execute`